### ISSUE TITLE
Included text of a module (Scrip. verses) does not refresh after editing the text, until Module is viewed in Output view

### ISSUE THREAD

**Issue by KimB2017** [sillsdev/ptx2pdf#652]
PTXprint 2.1.13 in Wasta 20.04, PT-Linux 9.0. If I edit the Scripture text referenced in a Module, PTXprint does not update the text until I change the Module view to Output view (Standard or Preview). Nothing else triggers it: regenerate the SFM, print 2x, close PTXprint/reopen, move away from View+Edit and back to it.

If this is an unavoidable step to take (viewing Output in PT after editing the text), it would be helpful to have a note on the View+Edit page somewhere--I guess the logical place might be added to the popup on the Generate button?

I don't think an archive will help this time, but let me know if you want it.

**Comment by markpenny** [comment]
Hi Kim, It sounds to me as if Paratext is not actually updating/saving the module text until you force it to do so (by changing the view to Output view). I'm guessing/hoping that there is a good reason that Paratext works in this way -- presumably because if it had to re-create the output every time you changed/edited something, it would probably grind to a halt.

But, this isn't anything that PTXprint has control over, so I don't think we can help fix this issue.

**Comment by KimB2017** [comment]
That's what I figured. I only noticed this because I replaced the GEN 
file to format a different language in my "Modules-only" project. Then I 
couldn't update the sfm by the Generate button. It might be helpful to 
add a note/warning to the Tooltip for that button, but maybe it'd be 
more effective to place a remark in the PT Module file. Any number of 
small edits to the text might never be noticed until a book is printed 
without them!

*Kim Blewett*
Language Technology Consultant (LSS)
SIL Papua New Guinea

**Comment by mnjames** [comment]
I've struggled with this in the past, and so I have a few questions to try to nail down the PT and ptxprint logic.

When you say you're printing "from a module", inside of ptxprint are you choosing a "Book"--like XXA--or are you choosing a "Bible Module" and then navigating to `something.sfm`? The reason I ask is because I think the two systems work differently.

My understanding is that as you're editing the specification for a Bible Module, PT is saving that spec to the `Modules` folder in the original `something.sfm` file. If you edit the original scripture text, nothing happens immediately. As you've noticed, only _after_ you change views does the `94XXAZZZ.SFM` file get updated.

If I'm correct about that, then if ptxprint is pointing to XXA then it's necessary to update the module within PT using change view hack. But if ptxprint is pointing directly to the spec file (in `Modules\something.sfm`) then that update shouldn't be necessary.

Does that seem to line up with your experience?

**Comment by KimB2017** [comment]
Yes, I just now checked and you are right. If I point ptxprint to the Module spec file the update appears right away, as opposed to pointing to/printing the XXA book.

However, I have not yet been successful in printing from the Module option. It's getting better every time y'all fix an issue but it's not there yet.. Now with v 2.1.13 I can not get the pics to appear; space for each pic is inserted now, though! I could not get the Front Matter to "generate" in the View+Edit tab either, but just now I see that my customized Title and Credits pages are now present
[TTCBks22-BookWPics-kimPTXprintArchive_noPicsInModulePrintout.zip](https://github.com/sillsdev/ptx2pdf/files/8426205/TTCBks22-BookWPics-kimPTXprintArchive_noPicsInModulePrintout.zip)
.
